libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -205,10 +205,34 @@ struct CopyTranslate: public yul::ASTCopier
 
 }
 
+string IRGeneratorForStatementsBase::code() const
+{
+	return m_code.str();
+}
+
+std::ostringstream& IRGeneratorForStatementsBase::appendCode(bool _addLocationComment)
+{
+	if (
+		_addLocationComment &&
+		m_currentLocation.isValid() &&
+		m_lastLocation != m_currentLocation
+	)
+		m_code << sourceLocationComment(m_currentLocation, m_context) << "\n";
+
+	m_lastLocation = m_currentLocation;
```

Comment:
Shouldn't it only be updated if `_addLocationComment` is true? When is it ever false?

Reply:
We have one call in the derived class, I think the `define()` function that needs access without a location being added, that's the only case.

I think practically it makes not much difference because the cases were this is used usually don't change location between calls.

I'll add change it to be based on teh `if` anyway.

Reply:
https://github.com/ethereum/solidity/pull/11450/files#r655419579